src/storage/heap_file.c

```diff
@@ -2575,7 +2575,7 @@ heap_classrepr_dump (THREAD_ENTRY * thread_p, FILE * fp, const OID * class_oid,
       goto exit_on_error;
     }
 
-  classname = heap_get_class_name (thread_p, class_oid);
+  classname = or_class_name (&recdes);
```

Comment:
please notice that `classname` is to be freed, which is no longer correct. 

I also think you can change `if` as `assert (classname != NULL);`. Please consider :)